ibm/src/main/java/nl/nn/adapterframework/extensions/ibm/IMSSender.java

```diff
@@ -0,0 +1,210 @@
+	private static final String   IIH_HEADER_REPLY_FORMAT	= "MQIMSVS ";	// (8 pos)
+	private	static final String   IIH_HEADER_MFS_AUTH		= "        ";	// Password (8 pos)
+	private static final byte[]   IIH_HEADER_TRAN_INSTANCE	= new byte[16];	// Only relevant for conversation (16 pos)
+	private static final String   IIH_HEADER_TRAN_STATE		= " ";			// MQITS_NOT_IN_CONVERSATION (1 pos)
+	private static final String   IIH_HEADER_COMMIT_MODE	= "1";			// MQICM-SEND-THEN-COMMIT (1 pos)
+	private static final String   IIH_HEADER_SECURITY_SCOPE	= "C";			// MQISS-CHECK (1 pos)
+	private static final String   IIH_HEADER_RESERVED		= " ";			// Reserved (1 pos)
+	
+	// MQ Fields
+	private static final String MQC_MQFMT_IMS	= "MQIMS   ";	// (8 pos)
+	private static final int MQENC_NATIVE = 273;				// copied from com.ibm.mq.MQC in com.ibm.mq.jar
+	private static final int CCSID_ISO_8859_1 = 819;
+	
+	private String transactionCode;
+	
+	/**
+	 * The transaction code that should be added in the header, must be 8 characters
+	 */
+	@IbisDoc({"transaction code that should be added to the header, must be 8 characters", ""})
+	public String getTransactionCode() {
+		return transactionCode;
+	}
+	public void setTransactionCode(String transactionCode) {
+		this.transactionCode = transactionCode;
+	}
+
+	@Override
+	public void configure() throws ConfigurationException {
+		if (StringUtils.isEmpty(getTransactionCode())) {
+			throw new ConfigurationException("transactionCode must be specified");
+		}
+		if (StringUtils.length(getTransactionCode()) != 8) {
+			throw new ConfigurationException("transactionCode must be 8 positions, current code [" + getTransactionCode() + "] with length [" + StringUtils.length(getTransactionCode()) + "]");
+		}
+		super.configure();
+	}
+	
+	@Override
+	public Message createMessage(Session session, String correlationID, String message)
+			throws NamingException, JMSException {
+		
+		BytesMessage bytesMessage = null;
+		bytesMessage = session.createBytesMessage();
+		
+		setMessageCorrelationID(bytesMessage, correlationID);
+		
+		ByteArrayOutputStream bos = new ByteArrayOutputStream();
+		
+		try {
+			bos.write(IIH_HEADER_STRUCT_ID.getBytes());
+			bos.write(intToBytes(IIH_HEADER_VERSION));
+			bos.write(intToBytes(IIH_HEADER_LENGTH));
+			bos.write(intToBytes(IIH_HEADER_ENCODING));
+			bos.write(intToBytes(IIH_HEADER_CODECHARSET));
+			bos.write(IIH_HEADER_FORMAT.getBytes());
+			bos.write(intToBytes(IIH_HEADER_FLAGS));
+			bos.write(IIH_HEADER_LTERM_OR.getBytes());
+			bos.write(IIH_HEADER_MFS_MAPNAME.getBytes());
+			bos.write(IIH_HEADER_REPLY_FORMAT.getBytes());
+			bos.write(IIH_HEADER_MFS_AUTH.getBytes());
+			bos.write(IIH_HEADER_TRAN_INSTANCE);
+			bos.write(IIH_HEADER_TRAN_STATE.getBytes());
+			bos.write(IIH_HEADER_COMMIT_MODE.getBytes());
+			bos.write(IIH_HEADER_SECURITY_SCOPE.getBytes());
+			bos.write(IIH_HEADER_RESERVED.getBytes());
+			
+			byte[] data = message.getBytes();
```

Comment:
Sorry, had het meteen moeten aanwijzen; Hier ook de character set gebruiken. 
Eigenlijk altijd/overal wanneer er Strings naar bytes gaan, of andersom, dan moet je daarbij de characterset aangeven.
Het beste maak je ook een test, waarin je in je message diacrieten (éèï) en het euro teken (€) gebruikt, Die moeten dan netjes heen en weer gaan.

Reply:
€ is geen onderdeel van de ISO_8859_1, dus bij omzetten van de string naar bytes is deze al verloren.